src/Api/Response/ResponseParser.php

```diff
@@ -201,27 +205,60 @@ protected function parseProducts(SimpleXMLElement $data)
         return $products;
     }
 
-    /**
-     * @param SimpleXMLElement $data
-     * @return array|null
-     */
-    protected function parseSmartDidYouMean(SimpleXMLElement $data)
+    protected function parseQueryInfoMessage(HttpRequest $request, SimpleXMLElement $data): array
     {
+        if (empty($data->query)) {
+            return [];
+        }
+
         $originalQuery = isset($data->query->originalQuery) ? $data->query->originalQuery->__toString() : null;
         $didYouMeanQuery = isset($data->query->didYouMeanQuery) ? $data->query->didYouMeanQuery->__toString() : null;
         $currentQuery = isset($data->query->queryString) ? $data->query->queryString->__toString() : null;
         $queryStringType = isset($data->query->queryString->attributes()->type)
             ? $data->query->queryString->attributes()->type->__toString()
             : null;
 
-        if ((empty($originalQuery) && empty($didYouMeanQuery)) || $queryStringType === 'forced') {
-            return null;
-        }
+        $requestParams = (array) $request->all();
 
         return [
-            'type' => !empty($didYouMeanQuery) ? 'did-you-mean' : $queryStringType,
-            'alternative_query' => !empty($didYouMeanQuery) ? $didYouMeanQuery : $currentQuery,
-            'original_query' => !empty($didYouMeanQuery) ? $currentQuery : $originalQuery
+            'originalQuery' => $originalQuery,
+            'didYouMeanQuery' => $didYouMeanQuery,
+            'currentQuery' => $currentQuery,
+            'queryStringType' => $queryStringType,
+            'selectedCategoryName' => $this->getSelectedCategoryName($requestParams),
+            'selectedVendorName' => $this->getSelectedVendorName($requestParams)
         ];
     }
+
+    /**
+     * @param array $requestParams
+     * @return string|null
+     */
+    private function getSelectedCategoryName(array $requestParams)
+    {
+        if (!$selectedCategory = isset($requestParams['attrib']['cat']) ? $requestParams['attrib']['cat'][0] : null) {
```

Comment:
Why are we still checking for falsiness inside `$requestParams['attrib']['cat'][0]`?

Reply:
I think it's cleaner to say `$selectedCat = $requestParams['attrib']['cat'][0] ?? null`.